### PR TITLE
Add versioning to devhub

### PR DESCRIPTION
**Pull request by JoshOrndorff**
This is a first draft attempt at adding versioning to devhub following the [docusaurus instructions](https://docusaurus.io/docs/en/versioning).

Remaining tasks / discussions to have before merging
* [x] How does [versioning interact with translations](https://docusaurus.io/docs/en/versioning#versioning-and-translations)
* [ ] How to version content outside of docusaurus (like recipes, kitties, etc)
* [x] What is the process for adding missing docs. Here is [a proposal](https://github.com/substrate-developer-hub/substrate-developer-hub.github.io/issues/118#issuecomment-507384480)

### CODE REVIEW

**Comment by kaichaosun** [comment]
Support version in translations is straightforward, the doc details [here](https://docusaurus.io/docs/en/translation#versioned-translations).

File: website/versioned_docs/version-1.0.0/contracts/cargo-contract.md [hunk 1]
```diff
@@ -0,0 +1,143 @@
+---
+title: Cargo Contract: The ink! CLI
```
Comment (kaichaosun) [inline website/versioned_docs/version-1.0.0/contracts/cargo-contract.md]: The versioned doc has removed the quotes, it caused the related translation doc to be:
```
---
title:
  Cargo Contract: The ink! CLI
id: version-1.0.0-cargo-contract
original_id: cargo-contract
---
```

Comment (kaichaosun) [reply website/versioned_docs/version-1.0.0/contracts/cargo-contract.md]: @JoshOrndorff and I have a discussion, and we decided to add the quotes to solve it.

Comment (JoshOrndorff) [reply website/versioned_docs/version-1.0.0/contracts/cargo-contract.md]: I now realize that cutting a version is dropping the quotes from titles in _all_ cases. We only notice it in the ones that have colons (`:`). Maybe a better solution is to not put colons in the title?

**Comment by kaichaosun** [comment]
There is one issue when I'm testing:
* The translation for doc title is not working in the versioned (v1.0.0) doc.
* With `yarn start`, If the title in the translated doc is not the same with the title in source doc, 
Chrome can't access the translation doc,
Firefox fallback to use the title in latest doc, the doc is accessible. 

This is not a blocker for us to process the work of version enable.
* I will test the GitHub page when it's merged, to see if there is the same issue.
* I will do a walk around fix to make sure the title in translated docs is same with the source doc.

I suggest that:
* Agree and fix the colons `:` in the title of source docs.
* Refresh this PR to cut a new v1.0.0 since there are other changes happened in recent a few days.
* Once add or modify docs, we need to make sure both latest and v1.0.0 are treated thoroughly once this PR gets merged.

Are there any other concerns from @JoshOrndorff @shawntabrizi ?

**Comment by kaichaosun** [comment]
I updated the source doc's title in this [commit](https://github.com/substrate-developer-hub/substrate-developer-hub.github.io/pull/157/commits/6288af7033ef2f8732e4dfaf4bb4d2d14cd48f39).
@JoshOrndorff Are you able to cut the new v1.0.0 version with master changes merged?

**Comment by JoshOrndorff** [comment]
I've merged the latest changes from `source` branch into this PR and re-cut the 1.0.0 version.

My last two remaining thoughts

* How will non-docusaurus content be versioned?
  * Rustdocs has it's own versioning system. Other items like `/recipes` should indicate which version they build against or implement their own versioning scheme
* How does this work with search results?
  * We can do [the same trick](https://github.com/substrate-developer-hub/substrate-developer-hub.github.io/blob/source/website/siteConfig.js#L139) we did for language filtering, but we are having problems with that filtering and the rustdocs.

IMO This ca nbe merged, and we sort out the search results in the existing issues about filtering search results.

**Comment by kaichaosun** [comment]
The Crowdin project exceeds the current plan, I'm figuring out what's the best option to solve it. If no, I may revert this PR.

**Pull request merged**